controlpanel/api/models/tool.py

```diff
@@ -204,3 +204,68 @@ def restart(self, id_token):
         """
 
         cluster.ToolDeployment(self.user, self.tool).restart(id_token)
+
+
+class HomeDirectoryManager:
+    """
+    Emulates a Django model manager.
+    """
+
+    def create(self, *args, **kwargs):
+        home_directory = HomeDirectory(*args, **kwargs)
+        home_directory.save()
+        return home_directory 
+
+    def filter(self, **kwargs):
+        return []
```

Comment:
I got rid of the `ToolDeploymentManager` as it was more confusing than anything.

I get that it's an abstraction but I found that make it look like it's a persisted model was more confusing than anything.

In this case is less confusing as we don't have any related models but I would still avoid this abstraction.

Reply:
Yeah... I did wonder about this. Just trying to "follow precedence". The manager object on a model instance is kinda expected in Django's ORM which is why I thought we had the `ToolDeployManager` (I couldn't see any other reason for having it). If there isn't I'll remove the equivalent from this PR. :+1:

Thanks for clearing that up - I did wonder and it was on my list of questions to ask when I submitted this out of draft for proper review.

Reply:
No problem, I don't think it wasn't working or anything but from experience with the other `ToolDeploymentManager` class it was more confusing than anything (to me at least) as it was adding an additional layer of abstraction and tricked me to think it was something to do with a database record.

Have a play and see what feels simpler ☺️